packages/cli/src/info.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2021 Opstrace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { call, race, delay } from "redux-saga/effects";
+import createSagaMiddleware from "redux-saga";
+import { createStore, applyMiddleware } from "redux";
+
+import { smErrorLastResort, getKubeConfigForOpstraceClusterOrDie } from "./util";
+import * as cli from "./index";
+import { log, die, SECOND, parseVersionFromImageTag } from "@opstrace/utils";
+import {
+  STORAGE_KEY,
+  CONFIGMAP_NAME,
+  CONTROLLER_NAME,
+  LatestControllerConfigSchema,
+  latestInstallerMetadataConfigKey,
+} from "@opstrace/controller-config";
+import {
+  AppsV1Api,
+  CoreV1Api,
+  VersionApi,
+  VersionInfo,
+  V1ConfigMap,
+  V1Deployment,
+} from "@kubernetes/client-node";
+
+/**
+ * getClusterInfo reads Kubernetes and Opstrace cluster state
+ * from the apiserver then prints this information in human-readable format.
+ */
+async function getClusterInfo() {
+  const kcfg = await getKubeConfigForOpstraceClusterOrDie(
+    cli.CLIARGS.cloudProvider,
+    cli.CLIARGS.instanceName,
+  );
+
+  // Read controller ConfigMap and die if not found.
+  let cm: V1ConfigMap;
+  try {
+    const coreClient = kcfg.makeApiClient(CoreV1Api);
```

Comment:
These kubeconfig API calls work, but is there a better library to use, like update reducers?

Reply:
The kubeconfig returned from [`getKubeConfigForOpstraceClusterOrDie`](https://github.com/opstrace/opstrace/blob/696f18892cf3ec621129c9054489105265233797/packages/cli/src/util.ts#L303) appears to just generate a kubeconfig dynamically using the same [underlying function](https://github.com/opstrace/opstrace/blob/696f18892cf3ec621129c9054489105265233797/packages/cli/src/util.ts#L348) as [`getKubecfgIfk8sClusterExists`](https://github.com/opstrace/opstrace/blob/main/packages/uninstaller/src/index.ts#L91-L105); will they not yield the same result? Perhaps exiting is not the right behavior though, and returning "undefined" for cluster version for an unavailable cluster using the function you suggested is desirable instead. Thoughts?

Reply:
If we can't get the kubeconfig to access the cluster, then we might as well fail early. wdyt?

Reply:
Yep agreed! That function is being used above to instantiate `kcfg`.